src/test/java/org/jabref/model/FieldChangeTest.java

```diff
@@ -0,0 +1,63 @@
+package org.jabref.model;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class FieldChangeTest {
+
+  private FieldChange Fieldchange;
+
+  @Test
+  void testHashCode() {
+    FieldChange fcNull = new FieldChange(null, null, null, null);
+    assertEquals(923521, fcNull.hashCode());
+  }
+
+  @Test
+  void testEquals() {
+    BibEntry entry = new BibEntry();
+    BibEntry entryOther = new BibEntry();
+    final Field field = StandardField.DOI;
+    final Field fieldOther = StandardField.DOI;
+    entry.setField(StandardField.DOI, "foo");
+    final String oldValue = "foo";
+    final String newValue = "bar";
+    final String oldValueOther = "fooX";
+    final String newValueOther = "barX";
+
+    FieldChange fc = new FieldChange(entry, field, oldValue, newValue);
+    FieldChange fcOther = new FieldChange(entryOther, fieldOther, oldValueOther, newValueOther);
+    FieldChange fcBlankAll = new FieldChange(null, null, null, null);
+    FieldChange fcBlankField = new FieldChange(entry, null, oldValue, newValue);
+    FieldChange fcBlankOldValue = new FieldChange(entry, field, null, newValue);
+    FieldChange fcBlankNewValue = new FieldChange(entry, field, oldValue, null);
+
+    assertFalse(fc.equals("foo"));
```

Comment:
assertEquals/assertNotEquals  already calls the underlying equals method of the object, so so need to test the equals with true/false